test/join.jl

```diff
@@ -325,4 +325,50 @@ module TestJoin
         @test all(isa.(o(on).columns,
                        [CategoricalVector{Union{T, Null}} for T in (Int, Float64)]))
     end
+
+    @testset "maintain Categorical levels ordering on join - non-`on` cols" begin
+        A = DataFrame(a = [1, 2, 3], b = ["a", "b", "c"])
+        c = levels!(categorical(["a", "b", "b"]), ["b", "a"])
+        B = DataFrame(b = ["a", "b", "c"], c = c)
+        @test levels(join(A, B, on=:b, kind=:inner)[:c]) == ["b", "a"]
```

Comment:
It would be nice to test `isordered` too, since that's easy to get wrong. While you're at it, testing the actual contents of the resulting column wouldn't hurt either, if that's not too much additional code.